lib/index.ts

```diff
@@ -1971,4 +2188,91 @@ export class Worker {
 			clearInterval(this.cacheRefreshInterval);
 		}
 	}
+
+	public async disableDataCache() {
+		if (this.cacheRefreshInterval) {
+			clearInterval(this.cacheRefreshInterval);
+		}
+	}
+
+	/**
+	 * Creates an action-evaluate-trigger that will process the trigger-like objects
+	 *
+	 */
+	private async createEvaluateTriggerAction(
+		logContext: LogContext,
+		session: string,
+		currentTime: Date,
+		currentContract: Contract<
+			ContractData,
+			{ [key: string]: Array<Contract<ContractData, any>> }
+		> | null,
+		insertedContract:
+			| Contract<
+					ContractData,
+					{ [key: string]: Array<Contract<ContractData, any>> }
+			  >
+			| TypeContract,
+		options: {
+			actor: any;
+			originator: any;
+			attachEvents: any;
+			timestamp: string | number | Date;
+			reason: any;
+			eventType: any;
+			eventPayload: any;
+		},
+	) {
+		const sessionContract = await this.kernel.getContractById<SessionContract>(
+			logContext,
+			session,
+			session,
+		);
+		const actionContract =
+			this.typeContracts['action-request@1.0.0'] ||
+			(await this.kernel.getContractBySlug<ActionContract>(
+				logContext,
+				this.session,
+				'action-request@1.0.0',
+			));
+		return this.insertCard(
+			logContext,
+			this.session,
+			actionContract,
+			{
+				timestamp: currentTime.toISOString(),
+				actor: sessionContract!.data.actor,
+				originator: options.originator,
+			},
+			{
+				data: {
+					action: 'action-evaluate-triggers@1.0.0',
+					card: insertedContract.id,
+					actor: sessionContract!.data.actor,
+					context: logContext,
+					input: {
+						id: insertedContract.id,
+					},
+					epoch: currentTime.valueOf(),
+					timestamp: currentTime.toISOString(),
+					originator: options.originator,
+					arguments: {
+						session,
+						currentTime: currentTime.toISOString(),
+						currentContract,
```

Comment:
My slightly obscure comment about permissions in the spec refers to this action requiring you to duplicate the contract data inside the action-request. This denormalization is tolerable, but exposes some potential security problems, in that you might have permission to view the action request, *but not* the contract nested inside it.
This is the kind of problem that immutable contracts resolves, because you can replace the full contract with a versioned ref, however, this is a feature we don't have yet :(